File: frontend/src/types/analysis.ts
export type Recommendation = "strong_buy" | "buy" | "hold" | "sell" | "strong_sell";

export interface RSIData { value: number; signal: "oversold" | "neutral" | "overbought"; }
export interface MACDData { histogram: number; signal: "bullish_cross" | "bearish_cross" | "neutral"; }
export interface MAData { value: number; above: boolean; }

export interface TechnicalData {
  rsi: RSIData;
  macd: MACDData;
  ma20: MAData;
  ma50: MAData;
  ma200: MAData;
  fiftytwo_week_position: number;
  bullish_count: number;
  total_count: number;
}

export interface FundamentalData {
  pe_ratio: number | null;
  sector_pe_median: number | null;
  pe_verdict: "cheaper_than_avg" | "fair_value" | "expensive";
  market_cap_label: "Mega" | "Large" | "Mid" | "Small" | null;
  dividend_yield: number | null;
  beta: number | null;
  beta_verdict: "low" | "medium" | "high" | null;
  fiftytwo_week_range_verdict: "near_low" | "mid_range" | "near_high";
  verdict_text: string;
}

export interface PlainEnglishData {
  newbie: string;
  basic: string;
  pro: string;
}

export interface Analysis {
  ticker: string;
  score: number;
  recommendation: Recommendation;
  plain_english: PlainEnglishData;
  technical: TechnicalData;
  fundamental: FundamentalData;
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel
from typing import Literal


class RSIData(BaseModel):
    value: float
    signal: Literal["oversold", "neutral", "overbought"]


class MACDData(BaseModel):
    histogram: float
    signal: Literal["bullish_cross", "bearish_cross", "neutral"]


class MAData(BaseModel):
    value: float
    above: bool


class TechnicalData(BaseModel):
    rsi: RSIData
    macd: MACDData
    ma20: MAData
    ma50: MAData
    ma200: MAData
    fiftytwo_week_position: float
    bullish_count: int
    total_count: int


class FundamentalData(BaseModel):
    pe_ratio: float | None
    sector_pe_median: float | None
    pe_verdict: Literal["cheaper_than_avg", "fair_value", "expensive"]
    market_cap_label: Literal["Mega", "Large", "Mid", "Small"] | None
    dividend_yield: float | None
    beta: float | None
    beta_verdict: Literal["low", "medium", "high"] | None
    fiftytwo_week_range_verdict: Literal["near_low", "mid_range", "near_high"]
    verdict_text: str


class PlainEnglishData(BaseModel):
    newbie: str
    basic: str
    pro: str


class AnalysisResponse(BaseModel):
    ticker: str
    score: float
    recommendation: Literal["strong_buy", "buy", "hold", "sell", "strong_sell"]
    plain_english: PlainEnglishData
    technical: TechnicalData
    fundamental: FundamentalData
